src/openforms/formio/formatters/custom.py

```diff
@@ -21,3 +22,16 @@ class MapFormatter(FormatterBase):
     def format(self, component: Component, value: list[float]) -> str:
         # use a comma here since its a single data element
         return ", ".join((str(x) for x in value))
+
+
+class AddressNLFormatter(FormatterBase):
+
+    empty_values = ({},)
+
+    def format(self, component: Component, value: dict[str, str]) -> str:
```

Comment:
I think making `value` a `TypedDict` makes sense here to document the expected types, since if you put in any dict of str -> str mapping, you will get `KeyError` on the `.format` call:

```py
>>> "{foo}".format()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'foo'
```